client/state/selectors/get-editor-close-config.js

```diff
@@ -46,8 +46,9 @@ export default function getEditorCloseConfig( state, siteId, postType, fseParent
 		};
 	}
 
-	// Customer Home.
-	if ( doesRouteMatch( /^\/home\/?/ ) ) {
+	// If a user comes from Home or from a fresh page load (i.e. Signup),
+	// redirect to Customer Home.
+	if ( ! lastNonEditorRoute || doesRouteMatch( /^\/home\/?/ ) ) {
```

Comment:
I'd appreciate some eyes on this. Signup completion queues a page reload, so `lastNonEditorRoute` is empty. This would also be the case for any user navigating directly from a non-Calypso page (i.e. site front-end). @apeatling @ianstewart how do we feel about that?

Alternatively, this could probably be tackled with a query param from Signup.

Reply:
I don't think it's related to this change, but while testing this I noticed that the back button label isn't getting updated, however the URL _is_. I know @noahtallen fixed this up in https://github.com/automattic/wp-calypso/pull/37749 and I haven't managed to work out why it wouldn't be working now, but it looks like the value is the [one generated here](https://github.com/automattic/wp-calypso/blob/ed94ca6d8ea81627383e2845689ef19cf0c6e620/apps/full-site-editing/full-site-editing-plugin/full-site-editing/class-full-site-editing.php#L133), which I assume is the original value of the back button before we update it with the label in this function?

I'll keep on looking into this tomorrow if no-one beats me to it.

As for routing users to the Home if they've come from the site front-end, etc, I think it makes sense since it'll get them to the Site Tools and where they select "Add a page", "Write a blog post", etc.

Reply:
And if everyone else is happy with switching the default URL when there's no `lastNonEditorRoute`, then we just need to [fix up this test](https://github.com/automattic/wp-calypso/blob/dc7ee470968f9137e19fb3a57f7a8c86d7fe4b05/client/state/selectors/test/get-editor-close-config.js#L28) to expect a `/home` URL.